### ISSUE TITLE
Openstack compute provider issue with the use of os-tenant-network

### ISSUE THREAD

**Issue by chewdevs** [hashicorp/terraform#14574]
Hi,

I'm working with a cloud based on Huawei Fusionsphere. Openstack native APIs are exposed (nova and neutron).
With the openstack 

### Terraform Version
0.9.5

### Affected Resource(s)
Please list the resources as a list, for example:
- openstack_compute_instance_v2

### Terraform Configuration Files
```hcl
resource "openstack_compute_instance_v2" "test-openstack" {
  name = "test-openstack"
  image_id = "d9bd237a-c298-47a7-aca8-3664e3debb44"
  flavor_id = "t2.small"
  security_groups = ["sg-lpr"]
  availability_zone = "eu-west-0a"
  region = "eu-west-0"

  network {
        uuid = "f1024038-db2f-4aed-a74e-21ee2c8dd1f7"
  }

}
```

### Debug Output
<script src="https://gist.github.com/chewdevs/e83f6b02b114d9f890b7442bce82561d.js"></script>

### Panic Output
None

### Expected Behavior
The instance show be created.

### Actual Behavior
The apply failed when trying to get network from the GET https://#####/v2/31cace43ed5d4fcfbda957aa7dfd3240/os-tenant-networks

The access to the `os-tenant-networks` is forbidden for this openstack implementation.

This issue is related to GH-5293 and GH-5216.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Important Factoids
None

### References
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:
- GH-5293
- GH-5216

**Comment by jtopjian** [comment]
@chewdevs Thanks or reporting this. There's an existing issue (https://github.com/hashicorp/terraform/issues/5293) for getting this fixed up.

For the time being, if you specify _both_ the network uuid and network name in the instance's `network` block, things should work correctly.

I'm going to close this issue, but do let me know if that workaround does not work for you. :)

**Comment by chewdevs** [comment]
I tried this workaround but the `os-tenant-network` is still used.

The terraform configuration
```
resource "openstack_compute_instance_v2" "test-openstack" {
  name = "test-openstack"
  image_id = "d9bd237a-c298-47a7-aca8-3664e3debb44"
  flavor_id = "t2.small"
  security_groups = ["sg-lpr"]
  availability_zone = "eu-west-0a"
  region = "eu-west-0"

  network {
        uuid = "f1024038-db2f-4aed-a74e-21ee2c8dd1f7"
        name = "01441486-edd8-4aac-8071-2bf3b6de4155"
  }
}
```

and the terraform log: <script src="https://gist.github.com/chewdevs/a1427a74917215b1fa401ebdb6d8e372.js"></script>

**Comment by chewdevs** [comment]
@jtopjian Thanks for your reply. I tried this workaroung but the `os-tenant-network` is still used.

The terrafform configuration used:
```
resource "openstack_compute_instance_v2" "test-openstack" {
  name = "test-openstack"
  image_id = "d9bd237a-c298-47a7-aca8-3664e3debb44"
  flavor_id = "t2.small"
  # key_pair = "fgzx6022"
  security_groups = ["sg-lpr"]
  availability_zone = "eu-west-0a"
  region = "eu-west-0"

  network {
        uuid = "f1024038-db2f-4aed-a74e-21ee2c8dd1f7"
        name = "01441486-edd8-4aac-8071-2bf3b6de4155"
  }
}
```

And the terraform log: <script src="https://gist.github.com/chewdevs/a1427a74917215b1fa401ebdb6d8e372.js"></script>

**Comment by jtopjian** [comment]
@chewdevs That's a really bizarre response. Can you do me a favor and do:

```shell
$ token=$(openstack token issue -c id -f value)
$ curl -s -H "X-Auth-Token:$token" https://###########/v2/31cace43ed5d4fcfbda957aa7dfd3240/os-tenant-networks | python -mjson.tool
```

And let me know the output?

**Comment by chewdevs** [comment]
When I do this curl, the answer is html and not json so the output is:
`No JSON object could be decoded`

HTML answers (I removed not relevant information from html) :
```
<html>
<head>
<TITLE>Request not authorized</TITLE>
<style>
....
</style>
</head>
<body>
<br>
<h1>Your request is not authorized</h1>
<p>Please contact your administrator.<br></p>
</body>
</html>
```

In comparaison, a curl to `os-networks` return json:
```
$ curl -s -H "X-Auth-Token:$token" https://###########/v2/31cace43ed5d4fcfbda957aa7dfd3240/os-networks | python -mjson.tool
{
    "networks": [
        {
            "broadcast": null,
            "cidr": null,
            "cidr_v6": null,
            "dns1": null,
            "dns2": null,
            "gateway": null,
            "gateway_v6": null,
            "id": "84a0f0f6-61df-45c5-8895-2da8f7eb536c",
            "label": "0d075fe7-9a22-4281-8f02-1d315f47b215",
            "netmask": null,
            "netmask_v6": null
        },
        {
            "broadcast": null,
            "cidr": null,
            "cidr_v6": null,
            "dns1": null,
            "dns2": null,
            "gateway": null,
            "gateway_v6": null,
            "id": "9f2db5a9-1d1d-4ffe-9150-1a17cf5a1d15",
            "label": "48681ca8-0a59-42cb-85bb-95fd503f5073",
            "netmask": null,
            "netmask_v6": null
        },
        {
            "broadcast": null,
            "cidr": null,
            "cidr_v6": null,
            "dns1": null,
            "dns2": null,
            "gateway": null,
            "gateway_v6": null,
            "id": "f1024038-db2f-4aed-a74e-21ee2c8dd1f7",
            "label": "01441486-edd8-4aac-8071-2bf3b6de4155",
            "netmask": null,
            "netmask_v6": null
        },
        {
            "broadcast": null,
            "cidr": null,
            "cidr_v6": null,
            "dns1": null,
            "dns2": null,
            "gateway": null,
            "gateway_v6": null,
            "id": "0a2228f2-7f8a-45f1-8e09-9039e1d09975",
            "label": "admin_external_net",
            "netmask": null,
            "netmask_v6": null
        
    ]
}
```

It explains the line in the terraform log:
`2017/05/18 13:17:11 [DEBUG] plugin: terraform: openstack-provider (internal) 2017/05/18 13:17:11 [DEBUG] Not logging because OpenStack response body isn't JSON`

**Comment by jtopjian** [comment]
> It explains the line in the terraform log:

Right. The surprising part is that the cloud you are using isn't returning a 404 or proper error code like that. It's simply returning a blank body. I can get a patch for this soon.

Thanks for helping with this :)

**Comment by chewdevs** [comment]
Agree with you, the answer is not standard... :)

Thanks for your feedback !

**Comment by jtopjian** [comment]
@chewdevs I was able to break my own `os-tenant-networks` and I think #14704 should fix this. Feel free to try it out if you're familiar with building Terraform from source / pull requests. :)

**Comment by chewdevs** [comment]
@jtopjian I applied your patch and was able to deploy an instance so your fix is working :)

Thank you !

**Comment by jtopjian** [comment]
@chewdevs Awesome news :)

**Comment by jtopjian** [comment]
Fixed by #14704
